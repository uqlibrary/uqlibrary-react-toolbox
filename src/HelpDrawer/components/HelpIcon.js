import React from 'react';
import PropTypes from 'prop-types';

import IconButton from 'material-ui/IconButton';
import FontIcon from 'material-ui/FontIcon';

import 'HelpIcon.scss';

const propTypes = {
    title: PropTypes.string,
    text: PropTypes.any.isRequired,
    buttonLabel: PropTypes.string,
    tooltip: PropTypes.string,
    tooltipIconColor: PropTypes.string,
    onClick: PropTypes.func,
    inline: PropTypes.bool,
    style: PropTypes.object
};

const defaultProps = {
    inline: false,
    style: {},
    tooltip: 'Click for more information',
    tooltipIconColor: '#CCCCCC'
};

const HelpIcon = ({title, text, buttonLabel, tooltip, tooltipIconColor, onClick, inline, style}) => {
    const setDrawerContent = () => {
        onClick(title, text, buttonLabel);
    };

    const classNames = `form-help-icon ${inline ? 'inline' : ''}`;

    return (
        <div className={classNames} style={{...style}}>
            <IconButton tooltip={tooltip} tooltipPosition="bottom-center" onClick={setDrawerContent}>
                <FontIcon className="material-icons" color={tooltipIconColor}>help_outline</FontIcon>
            </IconButton>
        </div>
    );
};

HelpIcon.propTypes = propTypes;
HelpIcon.defaultProps = defaultProps;

export default HelpIcon;
