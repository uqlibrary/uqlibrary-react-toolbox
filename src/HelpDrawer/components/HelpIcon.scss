.form-help-icon {
  text-align: right;

  button {
    position: absolute;
    top: 20px;
    right: 0;
    margin-top: -16px !important;
  }

  &.inline {
    position: relative;
    display: inline-block;
    top: auto;
    vertical-align: middle;
  }
}
