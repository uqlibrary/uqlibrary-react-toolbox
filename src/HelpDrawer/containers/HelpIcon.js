import {connect} from 'react-redux';

import HelpIcon from '../components/HelpIcon';
import {show} from '../actions';

const HelpIconContainer = connect(undefined, dispatch => {
    return {
        onClick: (title, text, buttonLabel) => dispatch(show(title, text, buttonLabel))
    };
})(HelpIcon);

export default HelpIconContainer;
