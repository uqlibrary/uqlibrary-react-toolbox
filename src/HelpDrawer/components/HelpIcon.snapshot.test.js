jest.dontMock('./HelpIcon');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import HelpIcon from './HelpIcon';

function setup(title, text, buttonLabel) {
    const props = {
        title: title,
        text: text,
        buttonLabel: buttonLabel
    };

    return shallow(<HelpIcon {...props} />);
}

describe('HelpIcon snapshots tests', () => {
    it('renders menu', () => {
        const hdText = 'Integer mattis rutrum velit nec posuere. Quisque rhoncus quam elit, eu tincidunt diam feugiat eu. Quisque luctus luctus mauris faucibus ornare. Ut eu metus vitae est euismod gravida ac vel augue. Duis sapien massa, tempor id vulputate nec, auctor nec augue. Donec ullamcorper dignissim metus at vulputate. Mauris non magna enim. Quisque gravida libero augue, efficitur vestibulum leo porttitor id. Etiam quis nisi vehicula, eleifend turpis ut, accumsan nisi. Nam nec nibh auctor, placerat ex vel, rhoncus risus. In fermentum ex sit amet augue molestie sodales. Sed eleifend convallis dui id euismod. Nam eu turpis non mi facilisis euismod. Morbi congue et lectus sed tempus. Interdum et malesuada fames ac ante ipsum primis in faucibus.';

        const wrapper = setup('HelpDrawer Title', hdText, 'Close');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
