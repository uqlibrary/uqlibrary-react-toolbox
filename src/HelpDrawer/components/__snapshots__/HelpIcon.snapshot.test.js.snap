// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpIcon snapshots tests renders menu 1`] = `
<div
  className="form-help-icon inline"
  style={Object {}}
>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltipPosition="bottom-center"
    touch={false}
  >
    <FontIcon
      className="material-icons"
      color="#CCCCCC"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      help_outline
    </FontIcon>
  </IconButton>
</div>
`;
